wlm/src/main/java/ai/djl/serving/wlm/ModelInfo.java

```diff
@@ -621,6 +632,80 @@ private void configPerModelSettings(Path modelDir) throws IOException {
         }
     }
 
+    void checkAvailableMemory(Device device, Path modelDir) throws IOException {
+        Properties prop = getServingProperties(modelDir);
+        long requiredMemory = intValue(prop, "required_memory_mb", -1) * 1024L * 1024;
+        WlmConfigManager wlmc = WlmConfigManager.getInstance();
+        int defMemory = wlmc.getReservedMemoryMb();
+        long reservedMemory = intValue(prop, "reserved_memory_mb", defMemory) * 1024L * 1024;
+        if (requiredMemory <= 0) {
+            try (Stream<Path> walk = Files.walk(modelDir)) {
+                // estimate the memory to be 1.2x of file size
+                requiredMemory =
+                        walk.filter(Files::isRegularFile).mapToLong(ModelInfo::getFileSize).sum();
+                requiredMemory = requiredMemory * 12 / 10;
```

Comment:
Is 1.2x a valid multiplier for all models? I'm ok with this for now, but I think we determined this from our experiments with DS based models, so not sure how accurate it is for other models (as long as it doesn't severely underestimate the required memory it should be ok).